cloudinit/cmd/main.py

```diff
@@ -403,9 +402,27 @@ def _should_wait_on_network(
 
 
 def main_init(name, args):
-    deps = [sources.DEP_FILESYSTEM, sources.DEP_NETWORK]
+    """
+    Cloud-init 'init' stage is broken up into the following sub-stages
+
+    1. Ensure that the init object fetches its config without errors
+    2. Setup logging/output redirections with resultant config (if any)
+    3. Initialize the cloud-init filesystem
+    4. Check if we can stop early by looking for various files
+    5. Fetch the datasource
+    6. Connect to the current instance location + update the cache
+    7. Consume the userdata (handlers get activated here)
+    8. Construct the modules object
+    9. Adjust any subsequent logging/output redirections using the modules
+       objects config as it may be different from init object
+    10. Run the modules for the 'init' stage
+    11. Done!
+    """
+
     if args.local:
-        deps = [sources.DEP_FILESYSTEM]
+        stage = stages.local
```

Comment:
`stage` holds all of the (formerly) magic strings as well as deps - this is all statically defined data that is known prior to runtime